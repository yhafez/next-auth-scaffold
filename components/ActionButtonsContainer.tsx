// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
