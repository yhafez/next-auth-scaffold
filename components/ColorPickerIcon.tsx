// Path: ./components/ColorPicker.tsx
import { useState, MouseEvent, useCallback, useEffect } from 'react'
import { Box, IconButton, Typography } from '@mui/material'
import { ColorLens } from '@mui/icons-material'
import { darken, lighten } from 'color2k'

import { ColorPickerPopover } from '.'
import { useBoundStore } from '../store'
import {
	getContrastColor,
	getSecondaryColor,
	mixColors,
	findOptimalOverlayOpacity,
	isOverlayNecessary,
} from '../utils/helpers'

export interface ColorPickerProps {
	name: string
	selectedInit?: boolean
}

export default function ColorPickerIcon({ name, selectedInit }: ColorPickerProps) {
	const { darkMode, customPalette, setCustomPalette } = useBoundStore()

	const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
	const [selected, setSelected] = useState(selectedInit ?? false)

	const handleColorChange = useCallback((color: string) => {
		const contrast = getContrastColor(color)
		let mixedColorPrimaryMain = color
		if (isOverlayNecessary(color, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(contrast, overlayColor, color, 7)
			mixedColorPrimaryMain = mixColors(color, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryLight = lighten(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryLight, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryLight,
				7,
			)
			mixedColorPrimaryLight = mixColors(mixedColorPrimaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryDark = darken(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryDark, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryDark,
				7,
			)
			mixedColorPrimaryDark = mixColors(mixedColorPrimaryDark, overlayColor, overlayOpacity)
		}

		const secondaryColor = getSecondaryColor(mixedColorPrimaryMain)
		const secondaryColorContrast = getContrastColor(secondaryColor)
		let mixedColorSecondaryMain = secondaryColor
		if (isOverlayNecessary(secondaryColor, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				secondaryColor,
				7,
			)
			mixedColorSecondaryMain = mixColors(secondaryColor, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryLight = lighten(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryLight, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryLight,
				7,
			)
			mixedColorSecondaryLight = mixColors(mixedColorSecondaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryDark = darken(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryDark, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryDark,
				7,
			)
			mixedColorSecondaryDark = mixColors(mixedColorSecondaryDark, overlayColor, overlayOpacity)
		}

		const newPalette = {
			...customPalette,
			primary: {
				main: mixedColorPrimaryMain,
				contrastText: contrast,
				light: mixedColorPrimaryLight,
				dark: mixedColorPrimaryDark,
			},
			secondary: {
				main: mixedColorSecondaryMain,
				contrastText: secondaryColorContrast,
				light: mixedColorSecondaryLight,
				dark: mixedColorSecondaryDark,
			},
		}

		setCustomPalette(newPalette)
	}, [])

	useEffect(() => {
		const customPalette = localStorage.getItem('customPalette')
		if (customPalette) handleColorChange(customPalette)
	}, [handleColorChange])

	useEffect(() => {
		setSelected(selectedInit ?? false)
	}, [selectedInit])

	const open = Boolean(anchorEl)
	const id = open ? 'simple-popover' : undefined

	const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
		setAnchorEl(event.currentTarget)
	}

	const handleClose = () => {
		setSelected(false)
		setAnchorEl(null)
	}

	return (
		<>
			<Box
				id={`${name}-color-picker-container`}
				sx={{
					display: 'flex',
					flexDirection: 'column',
					alignItems: 'center',
					justifyContent: 'center',
					p: 0,
					height: '100%',
					width: '100%',
					cursor: 'pointer',
				}}
			>
				<IconButton
					id={`${name}-color-picker-button`}
					aria-describedby={id}
					aria-labelledby={`${name}-color-picker-label`}
					onClick={handleClick}
				>
					<ColorLens
						id={`${name}-color-picker-icon`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
							fontSize: '1.5rem',
						}}
						aria-labelledby={`${name}-color-picker-label`}
						role="img"
					/>
				</IconButton>
				<label htmlFor={`${name}-color-picker-button`}>
					<Typography
						id={`${name}-color-picker-label`}
						variant="body2"
						sx={{
							textAlign: 'center',
							fontWeight: 500,
							cursor: 'pointer',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					>
						Theme Color
					</Typography>
				</label>
			</Box>
			<ColorPickerPopover
				name={name}
				anchorEl={anchorEl}
				handleClose={handleClose}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'center',
				}}
				transformOrigin={{
					vertical: 'top',
					horizontal: 'center',
				}}
				handleColorChange={handleColorChange}
			/>
		</>
	)
}
