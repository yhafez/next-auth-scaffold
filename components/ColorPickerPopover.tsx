// Path: ./components/ColorPickerPopover.tsx

import { useState, useCallback } from 'react'
import { Popover, PopoverOrigin } from '@mui/material'
import { useBoundStore } from '../store'
import { ChromePicker } from 'react-color'

export interface ColorPickerPopoverProps {
	name: string
	open: boolean
	anchorEl: HTMLButtonElement | null
	handleClose: () => void
	handleColorChange: (color: string) => void
	anchorOrigin?: PopoverOrigin
	transformOrigin?: PopoverOrigin
}

const ColorPickerPopover = ({
	name,
	open,
	anchorEl,
	handleClose,
	handleColorChange,
	anchorOrigin,
	transformOrigin,
}: ColorPickerPopoverProps) => {
	const { customPalette } = useBoundStore()

	const [color, setColor] = useState(customPalette.primary.main)

	const handleChange = useCallback(
		(color: any) => {
			setColor(color.hex)
			handleColorChange(color.hex)
		},
		[handleColorChange],
	)

	return (
		<Popover
			id={`${name}-color-picker-popover`}
			open={open}
			anchorEl={anchorEl}
			onClose={handleClose}
			anchorOrigin={anchorOrigin}
			transformOrigin={transformOrigin}
			sx={{
				cursor: 'pointer',
				'& .MuiPopover-paper': {
					padding: 2,
					backgroundColor: 'background.paper',
					color: 'text.primary',
				},
			}}
		>
			<ChromePicker color={color} onChange={color => handleChange(color)} disableAlpha />
		</Popover>
	)
}

export default ColorPickerPopover
