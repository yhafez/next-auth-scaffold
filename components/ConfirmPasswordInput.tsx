// Path: ./components/ConfirmPasswordInput.tsx
import { Dispatch, SetStateAction, KeyboardEvent } from 'react'

import PasswordInput from './PasswordInput'

export interface ConfirmPasswordInputProps {
	name: string
	password: string
	setPassword: Dispatch<SetStateAction<string>>
	confirmPassword: string
	setConfirmPassword: Dispatch<SetStateAction<string>>
	disabled?: boolean
	handleEnter?: (e: KeyboardEvent) => void
}

export default function ConfirmPasswordInput({
	name,
	password,
	setPassword,
	confirmPassword,
	setConfirmPassword,
	disabled = false,
	handleEnter,
}: ConfirmPasswordInputProps) {
	return (
		<>
			<PasswordInput
				name={name}
				value={password}
				setValue={setPassword}
				disabled={disabled}
				handleEnter={handleEnter}
			/>
			<PasswordInput
				name={`${name}`}
				value={confirmPassword}
				setValue={setConfirmPassword}
				disabled={disabled}
				isConfirmPassword={true}
				handleEnter={handleEnter}
			/>
		</>
	)
}
