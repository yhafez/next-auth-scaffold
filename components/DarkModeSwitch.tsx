// Path: ./components/DarkModeSwitch.tsx
import { useEffect, useState } from 'react'
import { Box, Switch, Typography } from '@mui/material'
import { WbSunny, Brightness2 } from '@mui/icons-material'

import { useBoundStore } from '../store'

export interface DarkModeSwitchProps {
	name: string
}

export default function DarkModeSwitch({ name }: DarkModeSwitchProps) {
	const { darkMode, setDarkMode } = useBoundStore()
	const [selected, setSelected] = useState(false)

	const handleDarkMode = () => {
		localStorage.setItem('darkMode', JSON.stringify(!darkMode))
		setDarkMode(!darkMode)
	}

	useEffect(() => {
		const darkMode = localStorage.getItem('darkMode') === 'true'
		setDarkMode(darkMode)
	}, [])

	const handleEnter = (e: React.KeyboardEvent<HTMLButtonElement>) => {
		if (e.key === 'Enter') {
			handleDarkMode()
		}
	}

	return (
		<Box
			id={`${name}-dark-mode-switch-container`}
			onMouseEnter={() => setSelected(true)}
			onMouseLeave={() => setSelected(false)}
			onFocus={() => setSelected(true)}
			onBlur={() => setSelected(false)}
			sx={{
				textAlign: 'center',
				display: 'flex',
				flexDirection: 'column',
				justifyContent: 'center',
				alignItems: 'center',
				height: '100%',
				width: '100%',
			}}
		>
			<Switch
				id={`${name}-dark-mode-switch`}
				checked={darkMode}
				onChange={handleDarkMode}
				onKeyDown={handleEnter}
				inputProps={{
					'aria-checked': darkMode,
					'aria-labelledby': `${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`,
					role: 'switch',
				}}
				sx={{
					fontSize: '1.5rem',
					color: selected ? (darkMode ? 'primary.light' : 'primary.dark') : 'primary.contrastText',
					'& .MuiSwitch-switchBase': {
						color: 'primary.contrastText',
						'&.Mui-checked': {
							color: 'primary.contrastText',
						},
						'&.Mui-checked + .MuiSwitch-track': {
							backgroundColor: 'secondary.main',
						},
					},
				}}
				checkedIcon={
					<Brightness2
						id={`${name}-dark-mode-switch-icon-checked`}
						aria-hidden={darkMode ? 'false' : 'true'}
						aria-labelledby={`${name}-dark-mode-switch-label-dark`}
						role="img"
						sx={{
							fontSize: '1.5rem',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					/>
				}
				icon={
					<WbSunny
						id={`${name}-dark-mode-switch-icon-unchecked`}
						aria-hidden={darkMode ? 'true' : 'false'}
						aria-labelledby={`${name}-dark-mode-switch-label-light`}
						role="img"
						sx={{
							fontSize: '1.5rem',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					/>
				}
			/>
			<label htmlFor={`${name}-dark-mode-switch`}>
				<Typography
					id={`${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`}
					variant="body2"
					sx={{
						fontWeight: 500,
						color: selected
							? darkMode
								? 'primary.light'
								: 'primary.dark'
							: 'primary.contrastText',
						cursor: 'pointer',
					}}
				>
					{darkMode ? 'Dark' : 'Light'}
					<br /> Mode
				</Typography>
			</label>
		</Box>
	)
}
