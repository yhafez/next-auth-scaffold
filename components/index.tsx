// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
