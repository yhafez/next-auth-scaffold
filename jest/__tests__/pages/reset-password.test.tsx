// Path: ./jest/__tests__/pages/reset-password-.test.tsx
import { render, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'
import { SessionProvider } from 'next-auth/react'
import { ThemeProvider } from '@mui/material/styles'

import ResetPassword from '../../../pages/reset-password/[id]/[token]'
import { Layout } from '../../../components'
import { defaultTheme } from '../../../theme'

describe('ResetPassword', () => {
	let container: HTMLElement
	beforeEach(() => {
		act(() => {
			container = render(
				<SessionProvider
					session={{
						token: {
							user: {
								id: '1',
								email: 'test@example.com',
								emailVerified: null,
								image: 'test',
								isDeactivated: false,
								name: null,
								role: 'USER',
								createdAt: new Date(),
								updatedAt: new Date(),
								isNewUser: false,
							},
						},
						expires: '1d',
					}}
				>
					<ThemeProvider theme={defaultTheme}>
						<Layout name="test" pageTitle="test">
							<ResetPassword />
						</Layout>
					</ThemeProvider>
				</SessionProvider>,
			).container

			return container
		})
	})

	it('Should have no accessibility violations', async () => {
		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should match snapshot', () => {
		expect(container).toMatchSnapshot()
	})
})
