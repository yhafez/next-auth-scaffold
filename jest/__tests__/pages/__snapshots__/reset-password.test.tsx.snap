// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14umlvy-MuiButtonBase-root-MuiButton-root"
    id="test-skip-nav-link"
    tabindex="0"
    type="button"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-3so7zb-MuiTypography-root"
      id="test-skip-nav-link-text"
    >
      Skip to main content
    </p>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiBox-root css-114gqq5"
    id="test-layout-container"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-38ln4h-MuiPaper-root-MuiAppBar-root"
      id="test-app-bar"
    >
      <nav
        aria-label="header navigation"
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l1du5z-MuiToolbar-root"
        id="test-toolbar"
      >
        <a
          aria-label="Home"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fqdykp-MuiTypography-root-MuiLink-root"
          href="/"
          id="test-toolbar-title-link"
          title="Home"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-noWrap css-63r4vv-MuiTypography-root"
            id="test-toolbar-title"
          >
            Next Auth Scaffold
          </h1>
        </a>
        <ul
          class="MuiList-root MuiList-padding css-166iuh2-MuiList-root"
          id="test-toolbar-icons"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-16ztiij-MuiListItem-root"
            id="test-toolbar-icons-color-picker-item"
          >
            <div
              class="MuiBox-root css-1lo2k45"
              id="test-toolbar-color-picker-container"
            >
              <button
                aria-labelledby="test-toolbar-color-picker-label"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                id="test-toolbar-color-picker-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  aria-labelledby="test-toolbar-color-picker-label"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                  data-testid="ColorLensIcon"
                  focusable="false"
                  id="test-toolbar-color-picker-icon"
                  role="img"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <label
                for="test-toolbar-color-picker-button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-7xs8rl-MuiTypography-root"
                  id="test-toolbar-color-picker-label"
                >
                  Theme Color
                </p>
              </label>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9gmft3-MuiListItem-root"
            id="test-toolbar-icons-dark-mode-switch-item"
          >
            <div
              class="MuiBox-root css-f9wnac"
              id="test-toolbar-dark-mode-switch-container"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-41alfi-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    aria-checked="false"
                    aria-labelledby="test-toolbar-dark-mode-switch-label-light"
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    id="test-toolbar-dark-mode-switch"
                    role="switch"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="false"
                    aria-labelledby="test-toolbar-dark-mode-switch-label-light"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq4ggj-MuiSvgIcon-root"
                    data-testid="WbSunnyIcon"
                    focusable="false"
                    id="test-toolbar-dark-mode-switch-icon-unchecked"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <label
                for="test-toolbar-dark-mode-switch"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-191g3cx-MuiTypography-root"
                  id="test-toolbar-dark-mode-switch-label-light"
                >
                  Light
                  <br />
                   Mode
                </p>
              </label>
            </div>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="MuiBox-root css-cn70eu"
      id="test-layout-content"
      tabindex="-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14umlvy-MuiButtonBase-root-MuiButton-root"
        id="reset-password-skip-nav-link"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-3so7zb-MuiTypography-root"
          id="reset-password-skip-nav-link-text"
        >
          Skip to main content
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root css-114gqq5"
        id="reset-password-layout-container"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-38ln4h-MuiPaper-root-MuiAppBar-root"
          id="reset-password-app-bar"
        >
          <nav
            aria-label="header navigation"
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l1du5z-MuiToolbar-root"
            id="reset-password-toolbar"
          >
            <a
              aria-label="Home"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fqdykp-MuiTypography-root-MuiLink-root"
              href="/"
              id="reset-password-toolbar-title-link"
              title="Home"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 MuiTypography-noWrap css-63r4vv-MuiTypography-root"
                id="reset-password-toolbar-title"
              >
                Next Auth Scaffold
              </h1>
            </a>
            <ul
              class="MuiList-root MuiList-padding css-166iuh2-MuiList-root"
              id="reset-password-toolbar-icons"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-16ztiij-MuiListItem-root"
                id="reset-password-toolbar-icons-color-picker-item"
              >
                <div
                  class="MuiBox-root css-1lo2k45"
                  id="reset-password-toolbar-color-picker-container"
                >
                  <button
                    aria-labelledby="reset-password-toolbar-color-picker-label"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    id="reset-password-toolbar-color-picker-button"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-labelledby="reset-password-toolbar-color-picker-label"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                      data-testid="ColorLensIcon"
                      focusable="false"
                      id="reset-password-toolbar-color-picker-icon"
                      role="img"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <label
                    for="reset-password-toolbar-color-picker-button"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-7xs8rl-MuiTypography-root"
                      id="reset-password-toolbar-color-picker-label"
                    >
                      Theme Color
                    </p>
                  </label>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9gmft3-MuiListItem-root"
                id="reset-password-toolbar-icons-dark-mode-switch-item"
              >
                <div
                  class="MuiBox-root css-f9wnac"
                  id="reset-password-toolbar-dark-mode-switch-container"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-41alfi-MuiSwitch-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        aria-checked="false"
                        aria-labelledby="reset-password-toolbar-dark-mode-switch-label-light"
                        class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                        id="reset-password-toolbar-dark-mode-switch"
                        role="switch"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="false"
                        aria-labelledby="reset-password-toolbar-dark-mode-switch-label-light"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq4ggj-MuiSvgIcon-root"
                        data-testid="WbSunnyIcon"
                        focusable="false"
                        id="reset-password-toolbar-dark-mode-switch-icon-unchecked"
                        role="img"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <label
                    for="reset-password-toolbar-dark-mode-switch"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-191g3cx-MuiTypography-root"
                      id="reset-password-toolbar-dark-mode-switch-label-light"
                    >
                      Light
                      <br />
                       Mode
                    </p>
                  </label>
                </div>
              </li>
            </ul>
          </nav>
        </header>
        <main
          class="MuiBox-root css-cn70eu"
          id="reset-password-layout-content"
          tabindex="-1"
        >
          <div
            class="MuiBox-root css-1bghj2m"
            id="reset-password-modal-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-118nrrt-MuiTypography-root"
              id="reset-password-modal-title-word-0"
            >
              Reset 
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-118nrrt-MuiTypography-root"
              id="reset-password-modal-title-word-1"
            >
              Password
            </p>
            <div
              class="MuiBox-root css-7n00rz"
              id="reset-password-modal-notification-container"
            >
              <div
                class="MuiBox-root css-j4g8gb"
                id="reset-password-modal-error-container"
              >
                <svg
                  aria-hidden="true"
                  aria-label="error"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  id="reset-password-modal-error-icon"
                  role="img"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2fkb5s-MuiTypography-root"
                  id="reset-password-modal-error"
                >
                  Unauthenticated
                </p>
              </div>
            </div>
            <form
              class="MuiBox-root css-xo10vp"
              id="reset-password-modal-form"
              role="form"
            >
              <div
                class="MuiBox-root css-ueupuo"
                id="reset-password-buttons-container"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13dwpwp-MuiButtonBase-root-MuiButton-root"
                  id="reset-password-back-to-login-navigation-button"
                  tabindex="0"
                  type="button"
                >
                  Back to Login
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </form>
          </div>
        </main>
      </div>
    </main>
  </div>
</div>
`;
